floss/language/go/extract.py

```diff
@@ -0,0 +1,385 @@
+    # Extract strings from string table in .rdata section
+    # .data:00537B40                 dd offset unk_4A1E3C
+    # .data:00537B44                 dd    4
+    # .data:00537B48                 dd offset unk_4A21C2
+    # .data:00537B4C                 dd    6
+
+    if arch == "amd64":
+        alignment = 0x10
+        fmt = "<QQ"
+    else:
+        alignment = 0x8
+        fmt = "<II"
+
+    for addr in range(0, len(section_data) - alignment // 2, alignment // 2):
+        try:
+            curr = section_data[addr : addr + alignment]
+            s_off, s_size = struct.unpack(fmt, curr)
+
+            if not s_off and not (1 <= s_size < 128):
+                continue
+
+            s_rva = s_off - pe.OPTIONAL_HEADER.ImageBase
+
+            if not pe.get_section_by_rva(s_rva):
+                continue
+
+            binary_string = pe.get_string_at_rva(s_rva, s_size)
+
+            addr = pe.get_offset_from_rva(s_rva)
+
+            yield from decode_and_validate(binary_string, addr, min_length)
+
+        except Exception as e:
+            logger.error(f"Error: {e}")
+            raise
+
+
+def extract_longstrings(
+    pe: pefile.PE, section_data, section_va, min_length, pattern, arch, regex_offset: Optional[int] = 0
+) -> Iterable[StaticString]:
+    for m in pattern.finditer(section_data):
+        s_off = struct.unpack("<I", m.group("offset"))[0]
+        s_size = struct.unpack("<B", m.group("size"))[0]
+
+        if arch == "amd64":
+            # .text:000000000048FF87 48 8D 1D 51 DC 01 00                          lea     rbx, unk_4ADBDF
+            # .rdata:00000000004ADBDF unk_4ADBDF      db  42h ; B             ; DATA XREF: main_main+1E7↑o
+            # To obtain the correct offset, we need to add the bytes we receive, which are 00 01 DC 51, to the current offset.
+
+            s_rva = s_off + m.start() + section_va + regex_offset
+
+        elif arch == "386":
+            # .text:0048CED8 8D 05 3A 49 4A 00                             lea     eax, unk_4A493A
+            # .rdata:004A493A unk_4A493A      db  42h ; B             ; DATA XREF: main_main+1C8↑o
+            # The correct offset is determined by the bytes we receive, which are 00 4A 49 3A.
+
+            s_rva = s_off - pe.OPTIONAL_HEADER.ImageBase
+
+        if not pe.get_section_by_rva(s_rva):
+            continue
+        addr = pe.get_offset_from_rva(s_rva)
+
+        binary_string = pe.get_string_at_rva(s_rva, s_size)
+        yield from decode_and_validate(binary_string, addr, min_length)
+
+
+def extract_go_strings(
+    sample: pathlib.Path,
+    min_length,
+) -> Iterable[StaticString]:
+    """
+    Get Go strings from a PE file.
+    Reference: https://github.com/mandiant/flare-floss/issues/779
+    """
+
+    try:
+        pe = pefile.PE(sample)
+    except pefile.PEFormatError as err:
+        logger.debug(f"invalid PE file: {err}")
+        raise ValueError(f"Invalid PE file: {err}")
```

Comment:
```suggestion
        logger.debug(f"invalid PE file: {err}")
        return
```

seems reasonable to return no strings if they can't be decoded. no need to raise an exception? or would a caller be expected to do something special in this case?